### ISSUE TITLE
Plans on changing project structure, packaging, naming

### ISSUE THREAD

**Issue by actionless** [actionless/oomox#129]
Regarding name (https://github.com/actionless/oomox/issues/116) so far I am thinking of `themix` so it will be used in the examples below.

Packages:
`themix-full` (metapackage, depends on rest of the packages, replaces `oomox` package)
`themix-gui`
`themix-theme-oomox`
`themix-theme-materia`
`themix-icons-gnome-colors`
`themix-icons-archdroid`
`themix-export-spotify`
`themix-format-base16`

Executable names:
`themix`
`themix-cli` — not implemented yet
And for compatibility:
`oomox-gui` -> `themix`
`oomox-cli` -> `themix-cli-oomox`
and so on

I was thinking about having single `themix-cli` entrypoint and things like `themix-cli --theme oomox` and `themix-cli --theme materia` but anyway we need to have single cli executables for each theme plugin, to make them able to work independently without main gui app installed.

Also I am considering moving project to a dedicated github project to encourage more community participation since it won't look that much like a personal project.

To be done:

- [ ] `./packaging/install_plugin.sh`

- [ ] writing new PKGBUILDs

**Comment by alexarnaud** [comment]
I always plan to package Oomox into Debian repository. After lot of 
change of structure I have decided to wait a little bit to finalize this 
work. Do you think this change could be the last big one for the next year?

I also agree with you, it could be great to have a community repository.

Best regards and thanks for your great job,
Alex.

Le 14/03/2018 à 03:45, Yauhen Kirylau a écrit :

**Comment by actionless** [comment]
repository structure won't be affected, only packaging scripts, so if you are writing your own packaging scripts this change shouldn't affect you in any way except for the name change

**Comment by alexarnaud** [comment]
Le 15/03/2018 à 01:54, Yauhen Kirylau a écrit :
> repository structure won't be affected, only packaging scripts, so if 
> you are writing your own packaging scripts this change shouldn't affect 
> you in any way except for the name change

Right, but if you change the source name, Ill have to change the source 
package name and all package name and provide transitional package. It's 
not a big issue for downstream repository but I would prefer to have a 
proper package directly if push it to the official Debian repository to 
avoid transitional package. I mean transitional package the name package 
oomox will depends on the new name, oomox-cli will also depend on the 
new name etc. We couldn't drop package from the repository if people 
have installed with the previous name. We'll also change path and so 
provide symbolic links to keep things working for script. So IMO it's 
easier to push directly the right package with the right package name.

Best regards,
Alex.

**Comment by actionless** [comment]
yup, in that case it cuold be reasonable to wait for the name change

**Comment by actionless** [comment]
@AladW sorry for spamming you from here but mb you have some recommendation regarding packaging "schema" for arch? ie the way how app will be splitted into the packages and if it should have a common group for them

**Comment by AladW** [comment]
Having 10 PKGBUILDs on AUR with different sources is probably annoying unless you go with unsupported means of installation (i.e. AUR helpers). 

I'm not familiar with the build system you use here so don't know what should be split and where; but for a single PKGBUILD, I'd consider a git submodule approach both for git and release version. (cf. https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/pacutils)

**Comment by actionless** [comment]
so you think it's fine to leave it as is?
https://github.com/actionless/oomox/blob/master/packaging/arch/PKGBUILD#L14-L22

**Comment by actionless** [comment]
At the same time splitted packaging will help to remove dependency on Inkscape (which is needed only for Materia theme plugin).

**Comment by actionless** [comment]
@smurphos going to tag 1.6.1 quite soon (before end of the month)
before next planned release (1.7.0) i gonna move all the repos to the new organization

name will be changed for 1.8.0 or 2.0 (whichever of them will be next version after)

**Comment by actionless** [comment]
ok, release is done, this weekend, i guess, i'll start movnig the repos

**Comment by actionless** [comment]
@nana-4 i was also thinking for themix 2.0 line, to preserve the legacy of the old oomox name (without changing recognizable app icon silhouette), to vectorize this into a new icon (bitmap looks not fine on lower sizes):


![oomox_2_0_4_16](https://user-images.githubusercontent.com/1655669/50876543-e902a980-13cd-11e9-988e-090128006a80.png)  ![oomox_2_0_4_24](https://user-images.githubusercontent.com/1655669/50876545-ec963080-13cd-11e9-97be-375106115364.png)  ![oomox_2_0_4_48](https://user-images.githubusercontent.com/1655669/50876548-f029b780-13cd-11e9-8334-678ba4ad8d19.png)  ![oomox_2_0_4_64](https://user-images.githubusercontent.com/1655669/50876550-f324a800-13cd-11e9-8d23-99097cce7afe.png)  ![oomox_2_0_4_128](https://user-images.githubusercontent.com/1655669/50876551-f5870200-13cd-11e9-9c58-faad5065cc63.png) ![oomox_2_0_4_256](https://user-images.githubusercontent.com/1655669/50876554-f91a8900-13cd-11e9-903d-f11b09931cf6.png)

![oomox_2_0_4](https://user-images.githubusercontent.com/1655669/50876558-0172c400-13ce-11e9-91db-fac081b44988.png)

**Comment by nana-4** [comment]
Is that the monkey face? Or human? Why someone's face for Themix app icon?

I’m sorry to say, but IMHO it looks a bit too scary face as an icon...

**Comment by actionless** [comment]
that's an alien, which is related to word 'oomox'

i thought what having a mascot would give to a project more humanity, also i think you should have some skills to re-draw it in a more kawaii way :3  (here's screenshot from the original tv show just for the reference:  ![2019-01-09--1547054339_619x432_scrot](https://user-images.githubusercontent.com/1655669/50936827-937fd880-1471-11e9-952f-b3632e995ab6.png) )

**Comment by nana-4** [comment]
My impression of the idea of mascot:

- Looking at the Wikipedia of [List of computing mascots](https://en.wikipedia.org/wiki/List_of_computing_mascots), languages/platforms/services/OSs tend to have a mascot. They are abstract and tend to have a large community.
- Some specific tools/apps (e.g. GIMP, Pidgin, eMule) have mascots and they are used for icons directly, but I think that trend is obsolete in modern times.
- So I think having a mascot for "Themix Project" might be good, but I'm not a big fan of using him/her for Themix app icon.
- Copying the design based on the existing TV character may be a copyright infringement. (Though I'm not so sure because I'm not a law expert.)

**Comment by actionless** [comment]
regarding copyright concern i think it falls under fair usage in parody context, though you could mb came with a more abstract alien, less resembling the original reference but still having some far resemblance

**Comment by actionless** [comment]
also it just came up to me what mascot's name could be Themengi

**Comment by nana-4** [comment]
So, what about something like this? This is a silhouette only design inspired by [Xue](https://en.wikipedia.org/wiki/File:Xfce_logo-footprint.svg), the mascot of Xfce.

![image](https://user-images.githubusercontent.com/9556384/51032072-05772f80-15e2-11e9-98bf-5924a1aaf0ac.png)

_NOTE:_ This is a pre-alpha version. I share this just to know what you think of such a design.

**Comment by actionless** [comment]
yeah! the shape is already looking attractive

**Comment by b1scu1t** [comment]
I'm considering packing this for Void Linux again, but the package is quite large and I would have preferred the package be split into sub-packages on a 'as needed' basis.

A CLI would be pretty neat too, to export some base16 stuff.

Do you still have interest in the restructure?

**Comment by actionless** [comment]
yeah, after adding new icon themes the whole package is indeed very huge, so that would make sense

regarding CLI there is already a separate issue: https://github.com/themix-project/oomox/issues/202

**Comment by LaurentTreguier** [comment]
Since April 2019 I was forced to split icon themes into subpackages for RPM packaging, since the number of files actually seemed to reach a hard limit that RPM packages support.
For now what I've done is putting each icon/GTK theme in their own package, which oomox/themix depends on.

**Comment by actionless** [comment]
i was thinking to package all themes as optional dependencies for AUR

**Comment by actionless** [comment]
so far i'm thinking to split into those packages:

```
 - oomox
depends on: themix-full

 - themix-full
depends on: themix-gui themix-theme-arc themix-theme-oomox themix-theme-materia
  themix-icons-archdroid themix-icons-gnome-colors themix-icons-numix
  themix-icons-papirus themix-icons-suruplus themix-icons-suruplus-aspromauros
  themix-export-spotify themix-plugin-base16 themix-import-image

 - themix-gui
optional deps: themix-theme-arc themix-theme-oomox themix-theme-materia
  themix-icons-archdroid themix-icons-gnome-colors themix-icons-numix
  themix-icons-papirus themix-icons-suruplus themix-icons-suruplus-aspromauros
  themix-export-spotify themix-plugin-base16 themix-import-image

 - themix-theme-arc
 - themix-theme-oomox
 - themix-theme-materia

 - themix-icons-archdroid
 - themix-icons-gnome-colors
 - themix-icons-numix
 - themix-icons-papirus
 - themix-icons-suruplus
 - themix-icons-suruplus-aspromauros

 - themix-export-spotify

 - themix-plugin-base16

 - themix-import-image
```

**Comment by b1scu1t** [comment]
Looks good to me, I would assume that the theme plugin would be bundled with said theme package?

And the GUI would have to be modular?

**Comment by actionless** [comment]
yup

**Comment by b1scu1t** [comment]
Testing the split packages in the AUR; `themix-gui-git`,`themix-icons-gnome-colors-git` & `themix-theme-oomox-git` seems to work well thus far. 
The GUI notifies if its missing an icon or gtk theme, that's cool.

**Comment by actionless** [comment]
thanks for the feedback!
i hope to finish up all the packages during this week

**Comment by actionless** [comment]
finally managed with it, still remains some renaming of the files and so on

**Comment by b1scu1t** [comment]
Sorry I didn't respond immediately.
I've been getting some trouble building [resvg-git](https://aur.archlinux.org/packages/resvg-git/) for the 'Arc' and 'Material' packages. The builds always fail.

![image](https://user-images.githubusercontent.com/41207956/77236344-159a7f80-6b94-11ea-82db-ce4055068f6c.png)

The base16-plugin returns a blank window on export, also.

![image](https://user-images.githubusercontent.com/41207956/77236438-dc164400-6b94-11ea-991c-ffd13d0e151f.png)

The Oomox theme and icons themes work fine though, as well as the image import plugin.

**Comment by actionless** [comment]
why resvg-git, not resvg?

please update `themix-plugin-base16-git` and try again

**Comment by actionless** [comment]
anyway, i've updated `resvg` AUR package, hope now it gonna work fine

**Comment by b1scu1t** [comment]
I've used `resvg-git` in the past to avoid the heap of KDE's dependencies (almost the whole KDE desktop) that was required to build the `resvg` package...I wish `resvg-bin` existed in the AUR.
`themix-plugin-base16-git` works perfectly after the update, thanks!

**Comment by actionless** [comment]
> to avoid the heap of KDE's dependencies

you can just remove kio from makedeps and the lines from the PKGBUILD which build the kde extension

**Comment by b1scu1t** [comment]
Finally got the `resvg` dependency satisfied with either `resvg-git` or `resvg-cairo`. 
I can export Materia themes just fine.

On an unrelated note:
The Arc theme I believe has a nasty bug that existed before `oomox` was split-up. 
Generating an Arc theme will progressively consume more RAM until the machine starts swapping aggressively. It never ends, `sassc` must be killed manually to stop it.

I'll create a separate issue for that...with better debug info too.

**Comment by actionless** [comment]
>The Arc theme I believe has a nasty bug 
>I'll create a separate issue for that...with better debug info too.

but that's a bug in Arc theme SCSS, not Themix/Oomox
